File: backend/verify_vonage_config.js
#!/usr/bin/env node
/**
 * Vonage API Key Configuration Verification Script
 * This script verifies that the VONAGE_API_KEY is properly configured and used throughout the application.
 */

const fs = require('fs');
const path = require('path');
const http = require('http');

function printHeader(message) {
    console.log('\n' + '='.repeat(60));
    console.log(`  ${message}`);
    console.log('='.repeat(60));
}

function printSuccess(message) {
    console.log(`✅ ${message}`);
}

function printError(message) {
    console.log(`❌ ${message}`);
}

function printInfo(message) {
    console.log(`ℹ️  ${message}`);
}

function checkEnvironmentVariables() {
    printHeader("Environment Variables Check");
    
    // Expected API key from Lovable/Supabase
    const expectedApiKey = "09bf89e3";
    
    // Check VONAGE_API_KEY
    const apiKey = process.env.VONAGE_API_KEY;
    if (apiKey) {
        if (apiKey === expectedApiKey) {
            printSuccess(`VONAGE_API_KEY is set correctly: ${apiKey}`);
        } else {
            printError(`VONAGE_API_KEY mismatch. Expected: ${expectedApiKey}, Got: ${apiKey}`);
        }
    } else {
        printError("VONAGE_API_KEY is not set");
    }
    
    // Check VONAGE_API_SECRET
    const apiSecret = process.env.VONAGE_API_SECRET;
    if (apiSecret) {
        if (apiSecret !== "your-vonage-api-secret-here") {
            printSuccess("VONAGE_API_SECRET is set");
        } else {
            printError("VONAGE_API_SECRET is using placeholder value");
        }
    } else {
        printError("VONAGE_API_SECRET is not set");
    }
    
    // Check other Vonage-related variables
    const vonageVars = [
        'VONAGE_VIDEO_API_KEY',
        'VONAGE_VIDEO_API_SECRET',
        'VONAGE_BRAND_NAME',
        'VONAGE_SENDER_ID'
    ];
    
    vonageVars.forEach(varName => {
        const value = process.env[varName];
        if (value) {
            printInfo(`${varName}: ${value}`);
        } else {
            printInfo(`${varName}: Not set (optional)`);
        }
    });
}

function checkConfigFiles() {
    printHeader("Configuration Files Check");
    
    const configFiles = [
        'config.env',
        'config.env.example',
        'src/config/enhanced_config.py',
        'src/routes/auth.py',
        'src/services/enhanced_vonage_service.py',
        'src/services/vonage_service.py'
    ];
    
    const expectedApiKey = "09bf89e3";
    
    configFiles.forEach(configFile => {
        const filePath = path.join(__dirname, configFile);
        if (fs.existsSync(filePath)) {
            try {
                const content = fs.readFileSync(filePath, 'utf8');
                if (content.includes(expectedApiKey)) {
                    printSuccess(`${configFile}: Contains correct API key`);
                } else if (content.includes('VONAGE_API_KEY')) {
                    printError(`${configFile}: Contains VONAGE_API_KEY but may not have correct value`);
                } else {
                    printInfo(`${configFile}: No VONAGE_API_KEY reference found`);
                }
            } catch (error) {
                printError(`${configFile}: Error reading file - ${error.message}`);
            }
        } else {
            printError(`${configFile}: File not found`);
        }
    });
}

function checkBackendHealth() {
    printHeader("Backend Health Check");
    
    return new Promise((resolve) => {
        const options = {
            hostname: 'localhost',
            port: 5000,
            path: '/health',
            method: 'GET',
            timeout: 5000
        };
        
        const req = http.request(options, (res) => {
            let data = '';
            res.on('data', (chunk) => {
                data += chunk;
            });
            
            res.on('end', () => {
                if (res.statusCode === 200) {
                    printSuccess("Backend server is running");
                    
                    try {
                        const healthData = JSON.parse(data);
                        const integrations = healthData.integrations || {};
                        const vonageStatus = integrations.vonage || false;
                        
                        if (vonageStatus) {
                            printSuccess("Vonage integration is connected");
                        } else {
                            printError("Vonage integration is not connected");
                        }
                    } catch (error) {
                        printError("Failed to parse health response");
                    }
                } else {
                    printError(`Backend server responded with status: ${res.statusCode}`);
                }
                resolve();
            });
        });
        
        req.on('error', (error) => {
            if (error.code === 'ECONNREFUSED') {
                printError("Backend server is not running");
            } else {
                printError(`Backend health check failed: ${error.message}`);
            }
            resolve();
        });
        
        req.on('timeout', () => {
            printError("Backend server request timed out");
            req.destroy();
            resolve();
        });
        
        req.end();
    });
}

function testVonageApiEndpoints() {
    printHeader("Vonage API Endpoints Test");
    
    const baseUrl = "http://localhost:5000/api/vonage";
    const endpoints = [
        "/start-verification",
        "/check-verification", 
        "/send-sms",
        "/account-balance"
    ];
    
    endpoints.forEach(endpoint => {
        const options = {
            hostname: 'localhost',
            port: 5000,
            path: `/api/vonage${endpoint}`,
            method: 'GET',
            timeout: 5000
        };
        
        const req = http.request(options, (res) => {
            if (res.statusCode === 200 || res.statusCode === 405) {
                printSuccess(`${endpoint}: Endpoint accessible`);
            } else {
                printError(`${endpoint}: Endpoint returned status ${res.statusCode}`);
            }
        });
        
        req.on('error', (error) => {
            printError(`${endpoint}: Endpoint not accessible - ${error.message}`);
        });
        
        req.on('timeout', () => {
            printError(`${endpoint}: Endpoint request timed out`);
            req.destroy();
        });
        
        req.end();
    });
}

function checkFrontendIntegration() {
    printHeader("Frontend Integration Check");
    
    const frontendFiles = [
        '../src/services/vonageApi.ts',
        '../src/services/smsVerificationService.ts'
    ];
    
    frontendFiles.forEach(filePath => {
        const fullPath = path.join(__dirname, filePath);
        if (fs.existsSync(fullPath)) {
            try {
                const content = fs.readFileSync(fullPath, 'utf8');
                if (content.includes('localhost:5000/api/vonage')) {
                    printSuccess(`${filePath}: Properly configured to use backend Vonage API`);
                } else {
                    printError(`${filePath}: May not be properly configured for Vonage API`);
                }
            } catch (error) {
                printError(`${filePath}: Error reading file - ${error.message}`);
            }
        } else {
            printError(`${filePath}: File not found`);
        }
    });
}

function generateSummary() {
    printHeader("Verification Summary");
    
    printInfo("This verification script checks that:");
    printInfo("1. VONAGE_API_KEY is set to '09bf89e3' (from Lovable/Supabase)");
    printInfo("2. Configuration files contain the correct API key");
    printInfo("3. Backend server is running and Vonage integration is connected");
    printInfo("4. Frontend is properly configured to use the backend Vonage API");
    
    printInfo("\nNext steps:");
    printInfo("1. Ensure VONAGE_API_SECRET is set to your actual Vonage API secret");
    printInfo("2. Test the Vonage integration with real phone numbers");
    printInfo("3. Monitor the backend logs for any Vonage API errors");
    printInfo("4. Verify that SMS verification and other Vonage features work correctly");
}

async function main() {
    printHeader("Vonage API Key Configuration Verification");
    printInfo(`Verification started at: ${new Date().toLocaleString()}`);
    
    // Run all verification checks
    checkEnvironmentVariables();
    checkConfigFiles();
    await checkBackendHealth();
    testVonageApiEndpoints();
    checkFrontendIntegration();
    generateSummary();
    
    printHeader("Verification Complete");
    printInfo("Review the results above to ensure proper Vonage API key configuration");
}

// Run the verification
main().catch(console.error);

File: backend/verify_vonage_config.py
#!/usr/bin/env python3
"""
Vonage API Key Configuration Verification Script
This script verifies that the VONAGE_API_KEY is properly configured and used throughout the application.
"""

import os
import sys
import requests
import json
from datetime import datetime

def print_header(message):
    """Print a formatted header"""
    print(f"\n{'='*60}")
    print(f"  {message}")
    print(f"{'='*60}")

def print_success(message):
    """Print a success message"""
    print(f"✅ {message}")

def print_error(message):
    """Print an error message"""
    print(f"❌ {message}")

def print_info(message):
    """Print an info message"""
    print(f"ℹ️  {message}")

def check_environment_variables():
    """Check if Vonage environment variables are properly set"""
    print_header("Environment Variables Check")
    
    # Expected API key from Lovable/Supabase
    expected_api_key = "09bf89e3"
    
    # Check VONAGE_API_KEY
    api_key = os.getenv('VONAGE_API_KEY')
    if api_key:
        if api_key == expected_api_key:
            print_success(f"VONAGE_API_KEY is set correctly: {api_key}")
        else:
            print_error(f"VONAGE_API_KEY mismatch. Expected: {expected_api_key}, Got: {api_key}")
    else:
        print_error("VONAGE_API_KEY is not set")
    
    # Check VONAGE_API_SECRET
    api_secret = os.getenv('VONAGE_API_SECRET')
    if api_secret:
        if api_secret != "your-vonage-api-secret-here":
            print_success("VONAGE_API_SECRET is set")
        else:
            print_error("VONAGE_API_SECRET is using placeholder value")
    else:
        print_error("VONAGE_API_SECRET is not set")
    
    # Check other Vonage-related variables
    vonage_vars = [
        'VONAGE_VIDEO_API_KEY',
        'VONAGE_VIDEO_API_SECRET',
        'VONAGE_BRAND_NAME',
        'VONAGE_SENDER_ID'
    ]
    
    for var in vonage_vars:
        value = os.getenv(var)
        if value:
            print_info(f"{var}: {value}")
        else:
            print_info(f"{var}: Not set (optional)")

def check_config_files():
    """Check configuration files for proper Vonage API key usage"""
    print_header("Configuration Files Check")
    
    config_files = [
        'backend/config.env',
        'backend/config.env.example',
        'backend/src/config/enhanced_config.py',
        'backend/src/routes/auth.py',
        'backend/src/services/enhanced_vonage_service.py',
        'backend/src/services/vonage_service.py'
    ]
    
    expected_api_key = "09bf89e3"
    
    for config_file in config_files:
        if os.path.exists(config_file):
            try:
                with open(config_file, 'r') as f:
                    content = f.read()
                    if expected_api_key in content:
                        print_success(f"{config_file}: Contains correct API key")
                    elif 'VONAGE_API_KEY' in content:
                        print_error(f"{config_file}: Contains VONAGE_API_KEY but may not have correct value")
                    else:
                        print_info(f"{config_file}: No VONAGE_API_KEY reference found")
            except Exception as e:
                print_error(f"{config_file}: Error reading file - {e}")
        else:
            print_error(f"{config_file}: File not found")

def check_backend_health():
    """Check if the backend is running and properly configured"""
    print_header("Backend Health Check")
    
    try:
        # Check if backend is running
        response = requests.get("http://localhost:5000/health", timeout=5)
        if response.status_code == 200:
            print_success("Backend server is running")
            
            # Check health response for Vonage integration status
            health_data = response.json()
            integrations = health_data.get('integrations', {})
            vonage_status = integrations.get('vonage', False)
            
            if vonage_status:
                print_success("Vonage integration is connected")
            else:
                print_error("Vonage integration is not connected")
                
        else:
            print_error(f"Backend server responded with status: {response.status_code}")
            
    except requests.exceptions.ConnectionError:
        print_error("Backend server is not running")
    except requests.exceptions.Timeout:
        print_error("Backend server request timed out")
    except Exception as e:
        print_error(f"Backend health check failed: {e}")

def test_vonage_api_endpoints():
    """Test Vonage API endpoints to verify API key usage"""
    print_header("Vonage API Endpoints Test")
    
    base_url = "http://localhost:5000/api/vonage"
    endpoints = [
        "/start-verification",
        "/check-verification", 
        "/send-sms",
        "/account-balance"
    ]
    
    for endpoint in endpoints:
        try:
            response = requests.get(f"{base_url}{endpoint}", timeout=5)
            if response.status_code in [200, 405]:  # 405 is Method Not Allowed (expected for GET on POST endpoints)
                print_success(f"{endpoint}: Endpoint accessible")
            else:
                print_error(f"{endpoint}: Endpoint returned status {response.status_code}")
        except requests.exceptions.RequestException as e:
            print_error(f"{endpoint}: Endpoint not accessible - {e}")

def check_frontend_integration():
    """Check if frontend is properly configured to use Vonage API"""
    print_header("Frontend Integration Check")
    
    frontend_files = [
        'src/services/vonageApi.ts',
        'src/services/smsVerificationService.ts'
    ]
    
    for file_path in frontend_files:
        if os.path.exists(file_path):
            try:
                with open(file_path, 'r') as f:
                    content = f.read()
                    if 'localhost:5000/api/vonage' in content:
                        print_success(f"{file_path}: Properly configured to use backend Vonage API")
                    else:
                        print_error(f"{file_path}: May not be properly configured for Vonage API")
            except Exception as e:
                print_error(f"{file_path}: Error reading file - {e}")
        else:
            print_error(f"{file_path}: File not found")

def generate_summary():
    """Generate a summary of the verification results"""
    print_header("Verification Summary")
    
    print_info("This verification script checks that:")
    print_info("1. VONAGE_API_KEY is set to '09bf89e3' (from Lovable/Supabase)")
    print_info("2. Configuration files contain the correct API key")
    print_info("3. Backend server is running and Vonage integration is connected")
    print_info("4. Frontend is properly configured to use the backend Vonage API")
    
    print_info("\nNext steps:")
    print_info("1. Ensure VONAGE_API_SECRET is set to your actual Vonage API secret")
    print_info("2. Test the Vonage integration with real phone numbers")
    print_info("3. Monitor the backend logs for any Vonage API errors")
    print_info("4. Verify that SMS verification and other Vonage features work correctly")

def main():
    """Main verification function"""
    print_header("Vonage API Key Configuration Verification")
    print_info(f"Verification started at: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    
    # Run all verification checks
    check_environment_variables()
    check_config_files()
    check_backend_health()
    test_vonage_api_endpoints()
    check_frontend_integration()
    generate_summary()
    
    print_header("Verification Complete")
    print_info("Review the results above to ensure proper Vonage API key configuration")

if __name__ == "__main__":
    main()
